src/FACT_internal.c

```diff
@@ -1689,8 +1680,20 @@ void FACT_INTERNAL_OnBufferEnd(FAudioVoiceCallback *callback, void* pContext)
 		1
 	);
 
-	/* Loop if applicable */
 	c->wave->streamOffset += buffer.AudioBytes;
+
+	/* Last buffer in the stream? */
+	if (	c->wave->streamOffset >= end &&
```

Comment:
Just noticed that this block can actually be squished into the block right below it, to `Loop/End if applicable`. LoopCount > 0 does the loop logic with Flags = 0, else sets FAUDIO_END_OF_STREAM.

Reply:
Yeah I noticed the similarities but didn’t want to make more changes without agreement on the approach. I’ll revert the assert and changes to the site of the callbacks.